<launch>
    
    <!-- Gazebo config -->
    <arg name="gazebo_simulation" default="true" />
    
    <!-- Controller config -->
    <arg name="controller" default="geometric" /> <!-- options: geometric, position -->
    
    <!-- Trajectory config -->
    <arg name="trajectory" default="fig8_10s"/>
    
    <!-- Controller node -->
    <node pkg="quad_control" name="mavros_cmd_node" type="mavros_cmd_node" output="screen">
        <param name="ctrl_mode" type="string" value="$(arg controller)" />
        <param name="enable_sim" value="$(arg gazebo_simulation)" />
        
        <param name="max_err_acc" value="10.0" />
        <!-- <param name="thrust_const" value="0.068" /> -->
        <!-- <param name="thrust_offset" value="0.045" /> -->
	<param name="thrust_const" value="0.06" />
        <param name="thrust_offset" value="0.1" />
        
        <param name="Kpos_x" value="16.0" />
        <param name="Kpos_y" value="16.0" />
        <param name="Kpos_z" value="16.0" />
        <param name="Kvel_x" value="5.6" />
        <param name="Kvel_y" value="5.6" />
        <param name="Kvel_z" value="5.6" />
        <param name="Katt_x" value="20" />
        <param name="Katt_y" value="20" />
        <param name="Katt_z" value="20" />
    </node>

    <!-- Trajectory node -->
    <node pkg="quad_control" name="trajectory_gen_node" type="trajectory_gen_node" output="screen">
        <param name="trajectory_file" type="string" value="$(find quad_control)/traj/traj_$(arg trajectory).csv" />
    </node>

</launch>

